client/src/Pages/BuyAndSellPage/index.js

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect, useState } from "react";
+import Plot from 'react-plotly.js';
+import { DateTime } from "luxon";
+// import { useHistory } from "react-router-history"
+/*
+TODO:
+[] Figure out cors
+[] 
+*/
+
+
+export default () => {
+
+    const companyID = "uber"
+
+    const [isLoading, setIsLoading] = useState(true)
+    const [data, setData] = useState()
+    const [companyDetails, setCompanyDetails] = useState()
+
+    const [value, setValue] = useState(0);
+
+    /**
+     * 
+     * @param {int} value Number of stocks in transaction
+     * @param {boolean} buy Is it a buy or a sell transaction
+     */
+    const onSubmitHandler = async (value, buy) => {
+        const res = await fetch("/buySell/", {
+            method: "POST",
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({ buy, "value": value })
+        })
+        const parsedResponse = await res.json()
+
+        if (parsedResponse.ok == false) {
+            alert(parsedResponse.error)
+        }
+        else {
+            alert("Transaction successful")
+            // return to previous page
+        }
+    }
+
+    useEffect(() => {
+        (async () => {
+            const response = await fetch("/buySell/?company=" + companyID)
+            const parsedResponse = await response.json()
+            // parsedResponse: {data: {time_fetched:int, trading_price:str}[], error: str}
+
+            const dates = parsedResponse.data.stockData.map(row => row.time_fetched).map(ts => {
+                return DateTime
+                    .fromSeconds(ts).setZone("America/New_York")
+                    .toFormat("kkkk-LL-dd HH:mm:ss")
+                // https://moment.github.io/luxon/#/formatting?id=table-of-tokens
+                // https://plotly.com/javascript/time-series/
+            })
+
+            const prices = parsedResponse.data.stockData.map(row => row.trading_price)
+
+            // console.log(date_data)
+            setData(() => ([{
+                x: dates,
+                y: prices,
+                type: 'scatter'
+            }]))
```

Comment:
Might wanna consider extrating this to a function for cleanliness sake

Reply:
Do you want me to extract the dates bit or dates and prices?

Reply:
I was thinking both probably, but totally up to you how you do it. Maybe both in 1 func. Maybe both in seperate funcs. I've just generally been told to keep what's in the body of the functional component to only what's necesarry e.g. hooks and return statements mostly